spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateSpecTest.java

```diff
@@ -0,0 +1,634 @@
+
+        assertNotEquals(oldCerts.get(0).getSerial().getId(), newCerts.get(0).getSerial().getId());
+    }
+
+    @Test
+    @DisplayName("should not allow consumer to regenerate another's certificates by entitlement")
+    public void shouldNotAllowConsumerToRegenerateAnothersCertsByEntitlement() throws Exception {
+        List<CertificateDTO> oldCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertThat(oldCerts).hasSize(1);
+        List<EntitlementDTO> ents = consumerApi.listEntitlements(system.getUuid());
+        ConsumerDTO system2 = client.consumers().register(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.trustedConsumer(system2.getUuid());
+        ConsumerClient consumerApi2 = consumerClient.consumers();
+        assertNotFound(
+            () -> consumerApi2.regenerateEntitlementCertificates(
+            system.getUuid(), ents.get(0).getId(), false));
+    }
+
+    @Test
+    @DisplayName("should not allow consumer to regenerate another's certificates")
+    public void shouldNotAllowConsumerToRegenerateAnothersCerts() throws Exception {
+        List<CertificateDTO> oldCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertThat(oldCerts).hasSize(1);
+        ConsumerDTO system2 = client.consumers().register(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.trustedConsumer(system2.getUuid());
+        ConsumerClient consumerApi2 = consumerClient.consumers();
+        assertNotFound(
+            () -> consumerApi2.regenerateEntitlementCertificates(system.getUuid(), null, false));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should be manually regenerated for a product")
+    public void shouldBeManuallyRegeneratedForAProductStandalone() throws Exception {
+        ProductDTO coolApp = Products.randomEng();
+        coolApp = ownerProductApi.createProductByOwner(owner.getKey(), coolApp);
+        pool = Pools.random(coolApp);
+        pool = ownerApi.createPool(owner.getKey(), pool);
+        consumerApi.bindPool(system.getUuid(), pool.getId(), 1);
+        List<CertificateDTO> oldCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertThat(oldCerts).hasSize(2);
+        AsyncJobStatusDTO result = entitlementsApi.regenerateEntitlementCertificatesForProduct(
+            coolApp.getId(), false);
+        if (result != null) {
+            AsyncJobStatusDTO status = jobApi.waitForJob(result.getId());
+            assertEquals("FINISHED", status.getState());
+        }
+
+        List<CertificateDTO> newCerts = consumerApi.fetchCertificates(system.getUuid());
+        assertEquals(oldCerts.size(), newCerts.size());
+        // System has two certs, but we only regenerated for one product, so the
+        // other serial should have remained the same:
+        assertThat(newCerts)
+            .filteredOn(oldCerts::contains)
+            .hasSize(1);
+    }
+
+    @Test
+    @OnlyInStandalone
```

Comment:
I know you've duplicated the tests and using the OnlyInStandalone/OnlyInHosted annotations in each version of the test because the ruby version was using the create_pool_and_subscription method, but we really don't need to do that.

We should only be using the OnlyInStandalone/OnlyInHosted when we are testing functionality that is different between the two modes (e.g. testing refresh, or locked vs custom pools/products, or hosted-only legacy bonus pools).

In the case for this spec test suite, we're just testing entitlement certs behaviour which should be the same in both hosted and standalone, so I'd say just remove the 'hosted' version of the tests,  just use the normal createPool/createProduct APIs, and remove the annotations so that the tests will run in both modes anyway